Windows make file support and reference material

charely6 opened the pull request:
I have updated the makefile to support windows
Added a new keytype
and added a bunch of reference material in its own folder for people to use when making their own key entires. I haven't made any others yet because I haven't needed to yet but I will as I need them or people contact looking for any.

charely6 committed:
added windows support and some reference material

Added windows support to the makefile
Added referenceMaterial folder with a bunch of pdfs that are lock smith reference material, Some are just manuals for key cutting machines but I think depth and spacing information can be extracted
added Key X103-KW12 shown in motorcycle_guide.pdf
added X103-KW12.scad following depth and spacing information from Framon_key_depth_guide_v8.pdf
Added more info to how to model key to try and help others going forward.
did some bug fixing on paths2openscad.py thought about using an updated version but the naming scheme changed and didn't want to have to redo everything in here.
I added another parameter to the keygen.py file called series_name that is referenced in the Framon_key_depth_guide_v8.pdf so the same "key blank" could be used for different spacing and depth systems.

charely6 committed:
More notes in how to model and a svg fix

I hadn't printed the actual key yet and realized that it was too wide by a new microns I successfully got it in by filing it down but wanted  to update it here. Note to do so I just modified the warding path in the svg using inkscape saved it and reran make clean and make.
I also updated X103-KW12.scad so the parse.py file would accept it. note it needs you to set a variable called name and variables called outlines_k and wardings_k (can't do them all in a row I don't really know why they made it that way)

Commit:
modified:   Makefile
	scad/best.gen.scad
	scad/kwikset.gen.scad
	scad/master.gen.scad
	scad/medeco.gen.scad
	scad/schlage.gen.scad

Commit:
updated the Makefile again I think this orignially might have run on
python2 but i am making it run on python3#

adventuredan commented:
thank you so much! i've been trying to get this program to work with the inkscape extension for years and the last few weeks ive really been trying to get it to work. addittionally i was about to put a pull request for the medeco classic scad to update the pin depth increments to .030" as it's set up with the biaxial .025" depth increment. i realized this after successfully printing several biaxial keys and then printing a classic key that i had to shave down the depths further with an exacto knife!

charely6 commented:
Hi, I'm always happy to see I'm not the only one working on a project. One of the things I ran into messing with updating the inkscape script thing is I think it rotated the output which broke everything for me. so I swapped it back.

What issues are we're you having with the inkscape extension? I would be up for helping you out with it.

adventuredan commented:
It was something to do with python2 I believe. I don't have the error code atm but the point is your updated code fixed it all! I've been using the original fork for years but never was able to successfully run the makefile to create the path polygons for custom keyways. These last few weeks I've been really bashing my head trying to figure it out and fix it (I know very little coding and was on the verge of learning python and scad from the ground up just to get it to work) when I came across your pull request and fork. Ran it today and for the first time was able to successfully create a custom keyway I've been trying to make for years. I was jumping up and down ecstatic! Really appreciate your work!

Next thing i need to figure out is how to implement the M3 slider! Seeing as how the patent ran out this year I believe the legal issues the original researcher was worried about are moot

charely6 commented:
well I'm happy I helped, what do you mean by M3 slider? is that a specific kind of cut key? or a specific key blank?

adventuredan commented:
M3 is the 3rd generation of medeco lock biaxial was 2nd generation and classic was the first. The only difference between a biaxial and an m3 lock and key is there is an extra bit of metal on the side of the key that depresses a button at the bottom right of the keyway which unlocks the slider. But the pin specs are identical to biaxial. You could even print a key using the biaxial code and key and then just depress that button with a paper clip. But would be fairly simple to add another variable I'm sure to include the m3 slider. It's literally just an extra rectangle on the side of the key.

charely6 commented:
> M3 is the 3rd generation of medeco lock biaxial was 2nd generation and classic was the first. The only difference between a biaxial and an m3 lock and key is there is an extra bit of metal on the side of the key that depresses a button at the bottom right of the keyway which unlocks the slider. But the pin specs are identical to biaxial. You could even print a key using the biaxial code and key and then just depress that button with a paper clip. But would be fairly simple to add another variable I'm sure to include the m3 slider. It's literally just an extra rectangle on the side of the key.

at that makes sense why you were annoyed you couldn't make new key gen files

check out for outline and stuff https://www.ilco.us/ilco-en/support/downloads/key-blank-directory

also I noticed I had found medeco cut info and put it in the new key reference material folder I added. 
I will probably add the ilco blank catalog as well, I want to try and computationally cut up the catalog, to automatically make svgs for all the various keys in there. I also want to look at python image stuff and see if I can come up with a key blank finder helper after I do that. I wanted to make a key for a toy capsule dispenser (haven't pushed all that quite yet) but I couldn't figure out the right blank for it.

adventuredan commented:
hey just an update! I wrote a new module for the medeco M3 slider and have it fully working! would love to merge it to your branch.

i did notice that something in your branch broke the schlage generation. was able to track it down to differences in the schlage.gen.scad.

not sure if it's something going on with the makefile causing it but i was able to swap the gen file with the original and it works fine.

charely6 commented:
awesome, If you want to make fork and do a pull request from my fork I'll review and merge it in. If you don't want to deal with that you could send me the the files or something and I can add them into my fork. I'll take a look at that schlage generation issue. I think I saw that also but I don't remember what I had changed to magically make it work.

charely6 commented:
I also turned on Discussions on my fork from here so we can chat someplace that is actually meant for it rather than here in a pull request.
https://github.com/charely6/keygen/discussions/2

charely6 committed:
Added more reference material and fixed schlage.svg for generating

By using an extension for inkscape called apply transforms I updated the schalge.svg file itself so when regenerated in the future it will generate the .gen.scad file with all the shapes in the correct orientation.

If this sort of issue comes again compare the .gen.scad file (specifically uncommenting the outling polygon at the bottom) to a know good one. make sure that the key looks to be in the same orientation to the known good one.

adventuredan committed:
Medeco Updates

Added M3.scad for generating Medeco M3 keys
Added m3_slider module to keygen.scad for cutting the M3 slider.
Updated the Medeco_Classic.scad to use the proper .030" cut depth increments instead of Biaxials' .025"
added L,C,R cut angles to Medeco_Biaxial.scad as these sometimes utilize the Classic center cuts. (possibly not in combination with KBQMDS?)
Added "M3" keyway to medeco.svg

adventuredan committed:
Merge pull request #1 from adventuredan/adventuredan-updates

Medeco Updates

adventuredan committed:
Add files via upload

adventuredan committed:
Add files via upload

adventuredan committed:
Delete keygen.scad

Accidentally uploaded to main folder

adventuredan committed:
Delete medeco.scad

adventuredan committed:
Delete medeco_M3.scad

adventuredan committed:
Delete medeco.svg

adventuredan committed:
Delete medeco_biaxial.scad

adventuredan committed:
Delete medeco_classic.scad

adventuredan committed:
Rename medeco_M3.scad.txt to medeco_M3.scad

adventuredan committed:
Add files via upload

adventuredan committed:
Delete scad/keygen.scad.txt

adventuredan committed:
Delete scad/medeco.scad.txt

adventuredan committed:
Delete scad/medeco_biaxial.scad.txt

adventuredan committed:
Delete scad/medeco_classic.scad.txt

adventuredan committed:
Add files via upload

adventuredan committed:
Add files via upload

adventuredan committed:
Delete scad/medeco.svg

adventuredan committed:
added comment notes for Biax and M3

added comment notes regarding Classic vs Biax cut positioning and some notes on Cam Lock versions

charely6 committed:
Merge pull request #3 from adventuredan/master

Medeco Updates
